internal/manifest/cr_deletion_check.go

```diff
@@ -0,0 +1,51 @@
+package manifest
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	declarative "github.com/kyma-project/lifecycle-manager/internal/declarative/v2"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewCustomResourceDeletionCheck creates a check that verifies that the Resource CR in the remote cluster is deleted.
+func NewCustomResourceDeletionCheck() *CRDeletionCheck {
+	return &CRDeletionCheck{}
+}
+
+type CRDeletionCheck struct{}
+
+func (c *CRDeletionCheck) Run(
```

Comment:
Please think about some consistent name, now it has `NewCustomResourceDeletionCheck` `CRDeletionCheck`, `WithCustomDeletionCheck`, `DeletionCheck` ... we can use one name.

Reply:
Among them, `CustomDeletionCheck` is misleading to me, basically, it's check if moduleCR gets deleted, please mention moduleCR at least.

Reply:
I disagree with mentioning at least module CR. Since this is an option in the declarative library and thus should be adaptable to some custom logic and not specific to only modules.

If we would remove/refactor the declarative library, we could get rid of all the options, because we anyway will only use it for the manifest reconciler.

But for now we are still sticking to it and I would not include module CR in the option name.
You can have a look at the other options, they are also named like this and the function passed into the options have the specific naming.

But I agree to have a consistent naming at least.

Reply:
As we discussed, we have a dedicated session for the declarative library.

And for the naming issue, I understand your concern, but still, if you look at the usage of this function `CustomDeletionCheck` - `options.DeletionCheck`, it's used in the `renderTargetResources` resources only, and the obj passing to it is manifest CR, in this case, it's not an option can be used anywhere. so when you decide the naming of this option, this should be aligned.